zzz_generated/app-common/app-common/zap-generated/attributes/Accessors.cpp

```diff
@@ -972,6 +972,36 @@ EmberAfStatus Set(chip::EndpointId endpoint, int16_t value)
 
 } // namespace HighTempThreshold
 
+namespace LowTempDwellTripPoint {
+
+EmberAfStatus Get(chip::EndpointId endpoint, uint32_t * value)
+{
+    return emberAfReadServerAttribute(endpoint, Clusters::DeviceTemperatureConfiguration::Id, Id, (uint8_t *) value,
+                                      sizeof(*value));
```

Comment:
This is not going to do the right thing....  It's reading as native-endian, and reading 4 bytes, not 3.  We clearly need better tests here somehow.

Reply:
You might be right. The call to `emberAfReadServerAttribute()` ends up in the `typeSensitiveMemCopy()` method, where the argument `readLength` is the value passed in the call to `emberAfReadServerAttribute` as the `sizeof(*value)`. In the case of `INT24U` this value is 3. But in the actual `memmove()` we see this:

```
        if (!ignoreReadLength && readLength < am->size)
        {
            return EMBER_ZCL_STATUS_INSUFFICIENT_SPACE;
        }
        if (src == NULL)
        {
            memset(dest, 0, am->size);
        }
        else
        {
            memmove(dest, src, am->size);
        }
```

This is the part of the code which we end up, when handling these types, which is not Strings, LongStrings or Lists.

The `readLength` is not used in this case, but just the attribute size it self `am->size`

Reply:
Hmm.  OK, that is a good point: the callee will only copy 3 bytes, and put them into our 4-byte space... where?  Unless there's some endianness-checking, it will put them in the "first 3 bytes", which will behave correctly for little-endian (but only if we make sure to zero out the uint32_t first) and not behave correctly for big-endian....

It might be clearer to read into an on-stack 3-byte buffer and then build the 32-bit int from that...